internal/internal_task_pollers.go

```diff
@@ -252,17 +248,25 @@ func (bp *basePoller) doPoll(
 	var err error
 	var result interface{}
 
-	doneC := make(chan struct{})
 	ctx, cancel, _ := newChannelContext(context.Background(), featureFlags, chanTimeout(pollTaskServiceTimeOut))
 
 	go func() {
+		defer cancel()
+		defer func() {
+			if p := recover(); p != nil {
+				bp.metricsScope.Counter(metrics.InternalPanicCounter).Inc(1)
+				st := getStackTraceRaw("base poller [panic]:", 7, 0)
+				bp.logger.Error("Unhandled panic.",
```

Comment:
```suggestion
				bp.logger.Error("Unhandled panic",
```
just because there's generally no punctuation like that in logs